src/game_events/manager.cpp

```diff
@@ -212,11 +119,24 @@ void manager::write_events(config& cfg) const
 
 void manager::execute_on_events(const std::string& event_id, manager::event_func_t func)
 {
-	iteration iter(event_id, *this);
+	// Start with a list of weak_ptrs to all matching event names (ie, "moveto").
+	handler_list matching_events = event_handlers_->get(event_id);
+
+	handler_list& var_list = event_handlers_->get_dynamic();
+
+	// Then check events with variables in their names for a match.
+	for(const auto iter = var_list.begin(); iter != var_list.end(); std::next(iter)) {
```

Comment:
This isn't how you iterate an STL container. In particular, `std::next()` doesn't change the iterator in-place, it returns a new iterator.

You should just use a range-for loop instead.